modules/javafx.controls/src/main/java/javafx/scene/control/skin/LabeledSkinBase.java

```diff
@@ -377,14 +377,20 @@ protected void updateChildren() {
                 labeled.getLineSpacing(), text.getBoundsType());
 
         // Now we want to add on the graphic if necessary!
-        double h = textHeight;
-        if (!isIgnoreGraphic()) {
-            final Node graphic = labeled.getGraphic();
-            if (contentDisplay == TOP || contentDisplay == BOTTOM) {
-                h = graphic.prefHeight(width) + gap + textHeight;
-            } else {
-                h = Math.max(textHeight, graphic.prefHeight(width));
-            }
+        double graphicHeight = graphic == null ? 0.0 :
+                Utils.boundedSize(graphic.prefHeight(-1), graphic.minHeight(-1), graphic.maxHeight(-1));
+
+        // Now add on the graphic, gap, and padding as appropriate
+        final Node graphic = labeled.getGraphic();
+        double height;
+        if (isIgnoreGraphic()) {
+            height = textHeight;
+        } else if (isIgnoreText()) {
+            height = graphicHeight;
+        } else if (contentDisplay == TOP || contentDisplay == BOTTOM){
+            height = graphic.prefHeight(-1) + gap + textHeight;
+        } else {
+            height = Math.max(textHeight, graphic.prefHeight(-1));
```

Comment:
The logic of this code seems to have changed more than just the fixing bug. Why are the `prefHeight` functions now called with `-1`?  Normally these should respect the content bias of the node involved.

Also, I notice that you check `graphic == null`, while the `isIgnoreGraphic` function checks quite a bit more:

    boolean isIgnoreGraphic() {
        return (graphic == null ||
                !graphic.isManaged() ||
                getSkinnable().getContentDisplay() == ContentDisplay.TEXT_ONLY);
    }

Doing all those operations on `graphic` when for example the `ContentDisplay` is `TEXT_ONLY` seems unnecessary.  Looking at the rest of the code, `graphicHeight` is only used in one branch in your if/elseif/elseif/else.

I also wonder if a simple fix like this would have the same result:

```diff
-       final double textHeight = Utils.computeTextHeight(font, cleanText,
+       final double textHeight = isIgnoreText() ? 0.0 : Utils.computeTextHeight(font, cleanText,
                 labeled.isWrapText() ? textWidth : 0,
                 labeled.getLineSpacing(), text.getBoundsType());
```

Reply:
> The logic of this code seems to have changed more than just the fixing bug. Why are the `prefHeight` functions now called with `-1`? Normally these should respect the content bias of the node involved.
> 

I used `-1` while calling `prefHeight` as I tried to make `computePrefHeight` in consistent with `computePrefWidth` method. But looking at the documentation in Node.java file, it looks like width should be used. So I made modifications to use width while calling `prefHeight`.


> Also, I notice that you check `graphic == null`, while the `isIgnoreGraphic` function checks quite a bit more:
> 
> ```
> boolean isIgnoreGraphic() {
>     return (graphic == null ||
>             !graphic.isManaged() ||
>             getSkinnable().getContentDisplay() == ContentDisplay.TEXT_ONLY);
> }
> ```
> 
> Doing all those operations on `graphic` when for example the `ContentDisplay` is `TEXT_ONLY` seems unnecessary. Looking at the rest of the code, `graphicHeight` is only used in one branch in your if/elseif/elseif/else.
> 

I carried forward the conditions which were already present. I believe removing the condition like `graphic == null` or `txt == null` from `isIgnoreGraphic()` or `isIgnoreText()` methods will create more combinations of conditions combined with ContentDisplay value during height calculation and we have to consider which all the height components to be considered in these cases. Since `graphic==null` is as good as considering  ContentDisplay.TEXT_ONLY, I feel it is better to keep this check.

I have updated the code to use `graphicHeight` in other cases of if/else branch.

> I also wonder if a simple fix like this would have the same result:
> 
> ```diff
> -       final double textHeight = Utils.computeTextHeight(font, cleanText,
> +       final double textHeight = isIgnoreText() ? 0.0 : Utils.computeTextHeight(font, cleanText,
>                  labeled.isWrapText() ? textWidth : 0,
>                  labeled.getLineSpacing(), text.getBoundsType());
> ```

This fixes only the issue of height calculation when text height needs to be ignored. But the original if/else branch present, considers text gap when not required. So I believe modification to the if/else branch is necessary.

Reply:
@hjohn please take a look and review the changes made.